scripts/maintenance/audit_branches.py

```
@@ -0,0 +1,101 @@
+        # 0. Check for shared history (merge-base) with origin/main
+        base = run_command(["git", "merge-base", "origin/main", branch])
+        if not base:
+            issues.append("Orphan branch (no shared history with main)")
+
+        # 1. Check for missing critical files using git ls-tree
+        branch_files_raw = run_command(["git", "ls-tree", "-r", "--name-only", branch])
+        if branch_files_raw:
+            branch_files = set(branch_files_raw.split('\n'))
+            for path in CRITICAL_FILES:
+                if path not in branch_files:
+                    issues.append(f"Missing {path}")
```

Ruff (0.15.10)

[warning] Use `list.extend` to create a transformed list

(PERF401)